src/Piral.Blazor.Core/ComponentActivationService.cs

```diff
@@ -80,22 +88,80 @@ public void DeactivateComponent(string componentName)
             }
         }
 
-        public void LoadComponentsFromAssembly(Assembly assembly)
+        public void LoadComponentsFromAssembly(Assembly assembly, IDictionary<string, object> args = default)

```

Comment:
Let's remove these and solve it as follows:

1. A codegen to bring in the `registerPage` calls by reading out the assembly's meta (being picky on the page attributes) and then bringing them in via a conversion of their route (blazor format to react router format) and calling `fromBlazor` with `page-${original-page-route}` as name.
2. The component resolver in loadcomponents does not only collect expose pilet attributes, but also page attributes registered as `page-${original-page-route}`